tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ServiceWrapperCommand.java

```diff
@@ -69,48 +71,90 @@ private File checkStaticWrapperFile() throws IOException
             {
                 if( lib.exists() && lib.getName().endsWith( "portal-kernel.jar" ) )
                 {
-                    targetJarFile.add( lib );
+                    portalkernelJar = lib;
+                    break;
                 }
             }
 
             libFiles = FileListing.getFileListing( new File( bundleServerPath.append( "../osgi" ).toOSString() ) );
+            libFiles.add( portalkernelJar );
 
-            for( File lib : libFiles )
+            if( !libFiles.isEmpty() )
             {
-                if( lib.exists() && lib.getName().endsWith( "api.jar" ) )
+                for( File lib : libFiles )
                 {
-                    targetJarFile.add( lib );
+                    if( lib.getName().endsWith( ".lpkg" ) )
+                    {
+                        try(JarFile jar = new JarFile( lib ))
+                        {
+                            Enumeration<JarEntry> enu = jar.entries();
+
+                            while( enu.hasMoreElements() )
+                            {
+                                JarEntry entry = enu.nextElement();
+
+                                String name = entry.getName();
+
+                                if( name.contains( ".api" ) )
+                                {
+                                    JarEntry jarentry = jar.getJarEntry( name );
+                                    InputStream inputStream = jar.getInputStream( jarentry );
+
+                                    JarInputStream jarInputStream = new JarInputStream( inputStream );
+                                    JarEntry nextJarEntry;
+
+                                    while( ( nextJarEntry = jarInputStream.getNextJarEntry() ) != null )
+                                    {
+                                        String entryName = nextJarEntry.getName();
+
+                                        if( entryName.endsWith( "ServiceWrapper.class" ) &&
+                                            !( entryName.contains( "$" ) ) )
+                                        {
+                                            entryName = entryName.replaceAll( "\\\\", "." ).replaceAll( "/", "." );
+                                            entryName = entryName.substring( 0, entryName.lastIndexOf( "." ) );
+                                            wrapperList.add( entryName );
+                                        }
+                                    }
+
+                                    jarInputStream.close();
```

Comment:
Don't manually close this stream, instead use another try/with/resources block for the since JarInputStream is autoClosable